File: ios/FurniturePlatform/Sources/Views/OfflineView.swift
import SwiftUI

/// Server umuman javob bermaganda (internet yo'q, backend o'chiq) ko'rsatiladigan
/// to'liq ekran holati — xom `URLError` matni o'rniga. Oddiy API xatolari
/// (400/401 va h.k.) buni ishlatmaydi, chunki ular server ishlab turganini
/// bildiradi — faqat `APIError.offline` uchun (qarang `isOffline(_:)`).
struct OfflineView: View {
    let onRetry: () -> Void
    var message: String? = nil

    static func isOffline(_ error: Error?) -> Bool {
        if case .offline = error as? APIError { return true }
        return false
    }

    var body: some View {
        VStack(spacing: 16) {
            Image(systemName: "wifi.slash")
                .font(.system(size: 44))
                .foregroundStyle(Color.brandDeep.opacity(0.5))
            Text("Internet aloqasi yo'q")
                .font(.headline)
            Text(message ?? "Serverga ulanib bo'lmadi. Internetingizni tekshirib, qayta urinib ko'ring.")
                .font(.subheadline)
                .foregroundStyle(.secondary)
                .multilineTextAlignment(.center)
                .padding(.horizontal, 32)
            Button(action: onRetry) {
                Label("Qayta urinish", systemImage: "arrow.clockwise")
                    .font(.subheadline).bold()
                    .padding(.horizontal, 18).padding(.vertical, 10)
                    .background(Color.brandDeep)
                    .foregroundStyle(Color.brandPrimary)
                    .clipShape(Capsule())
            }
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .padding(.top, 40)
    }
}

File: flutter_app/lib/widgets/offline_view.dart
import 'package:flutter/material.dart';
import '../api_client.dart';
import '../theme.dart';

/// Server umuman javob bermaganda (internet yo'q, backend o'chiq) ko'rsatiladigan
/// to'liq ekran holati — xom `SocketException`/`TimeoutException` matni o'rniga.
/// Oddiy API xatolari (400/401 va h.k.) buni ishlatmaydi, chunki ular server
/// ishlab turganini bildiradi — faqat [NetworkException] uchun.
class OfflineView extends StatelessWidget {
  final VoidCallback onRetry;
  final String? message;
  const OfflineView({super.key, required this.onRetry, this.message});

  /// `error` — catch bloklarida ushlangan xato obyekti. `NetworkException`
  /// bo'lsagina true qaytaradi, aks holda oddiy inline xato ko'rsatilishi kerak.
  static bool isNetworkError(Object? error) => error is NetworkException;

  @override
  Widget build(BuildContext context) {
    return Center(
      child: Padding(
        padding: const EdgeInsets.all(32),
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            Icon(Icons.wifi_off_rounded, size: 56, color: AppColors.deep.withValues(alpha: 0.5)),
            const SizedBox(height: 16),
            const Text(
              'Internet aloqasi yo\'q',
              style: TextStyle(fontSize: 17, fontWeight: FontWeight.w700),
              textAlign: TextAlign.center,
            ),
            const SizedBox(height: 6),
            Text(
              message ?? 'Serverga ulanib bo\'lmadi. Internetingizni tekshirib, qayta urinib ko\'ring.',
              style: const TextStyle(fontSize: 13.5, color: Color(0xFF8A7357)),
              textAlign: TextAlign.center,
            ),
            const SizedBox(height: 20),
            FilledButton.icon(
              onPressed: onRetry,
              style: FilledButton.styleFrom(backgroundColor: AppColors.deep),
              icon: const Icon(Icons.refresh_rounded),
              label: const Text('Qayta urinish'),
            ),
          ],
        ),
      ),
    );
  }
}
